### PR TITLE
Features/user signup

### PR DESCRIPTION
**Pull request by seanwu84**

### CODE REVIEW

File: public/js/user.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+"use strict";
```
Comment (bacbartlett) [inline public/js/user.js:1]: Why is this folder in public? I don't think we want it there

File: routes/api/users.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        const token = generateNewToken(user);
+        res.status(201).json({
+            user: { id: user.id },
+            token,
+        });
+    })
+);
+
+router.post(
```
Comment (bacbartlett) [inline routes/api/users.js:41]: This whole thing needs to be commented or else we will get an error

Comment (bacbartlett) [reply routes/api/users.js:41]: 41-61

File: routes/index.js [hunk 3]
```diff
@@ -5,4 +5,8 @@ router.get("/", (req, res) => {
   res.render("index");
 });
 
+router.get("/sign-up", (req, res) => {
```
Comment (bacbartlett) [inline routes/index.js:8]: I think we decided to make this route "/users/sign-up"

File: utils/utils.js [hunk 4]
```diff
@@ -1,10 +1,27 @@
-const csrf = require('csurf');
 
+const { validationResult } = require("express-validator");
+const csrf = require('csurf');
 const csrfProtection = csrf({ cookie: true });
 
 const asyncHandler = handler => (req, res, next) => handler(req, res, next).catch(next);
 
+const handleValidationErrors = (req, res, next) => {
```
Comment (bacbartlett) [inline utils/utils.js:8]: You tested this, ya? Just want to double check because I don't know how validator works

**Review by the-harry-higgins — APPROVED** [review]

**Pull request merged**